ARCHITECTURE.md

```diff
@@ -0,0 +1,197 @@
+- _Disrupt where disruption is necessary_ &mdash; the Nix ecosystem has a fairly rigid set of principles and norms that we don't think always apply in our use cases.
+- _Look left, right, above and beyond_ &mdash; our end-to-end perspective commands us to actively seek and reach out to other projects and ecosystems to compose our value chain; there's no place for the "not invented here"-syndrome.
+
+## Scope
+
+These are big goals and ideals.
+In the interest of practical advancements, we'll narrow down the scope in this section.
+
+We can subdivide (not break up!) our process into roughly three regions with different shapes and characteristics:
+
+- **Development Environment** roughly covers _code-to-commit_.
+- **Packaging Pipeline** roughly covers _commit-to-distribution_.
+- **Deployment and Beyond** roughly covers _distribution-to-next-rollout_.
+
+We delegate:
+
+- The **Development Environment** to a trusted project in the broader Nix Community employing community outreach to promote our cause and ensure it is at least not accidentally sabotaged.
+- The **Deployment and Beyond** by cultivating outreach and dovetailing with initiatives of, among others, the Cloud Native ecosystem.
+
+And we focus on:
+
+- The **Packaging Pipeline**
+- _Interfaces_ and _Integration_ with the other two
+
+## Architecture
+
+### Overview
+
+#### Locating Standard in the SDLC
+
+Where is Standard located in the big picture?
+
+This graphic locates Standard across the SDLC & ALM.
+
+But not only that.
+It also explains how automation in itself is implemented as _code_, just as the application itself.
+Therefore, we make a distinction between:
+
+- first order application code on the first level (L1); and
+- higher order supporting code on higher levels as exemplified by L2 and L3.
+
+> Glossary:
+>
+> _L2 & L3_ have no clearly defined meaning.
+> They represent that we may observe multiple layers of higher order code when automating.
+> Examples could be bash scripts, configuration data, platform utility code and more.
+
+<div align="center"><img src="./artwork/sdlc.svg" width="900" /></div>
+
+#### Standard's Components and their Value Contribution
+
+What is Standard made of? And how do its components contribute value?
+
+On the left side of the graphic, we show how Standard, like an onion, is build in layers:
+
+Center to Standard is [`divnix/paisano`](https://github.com/divnix/paisano).
+This flake (i.e. "Nix library") implements two main abstractions: Block Types and Cells.
+
+_**Block Types**_ are not unlike Rust's traits or Golang's interfaces.
+They are abstract definitions of artifact classes.
+Those abstract classes implement _shared functionality_.
+
+A few examples of artifact classes in our scope are: packages, containers, scripts and manifests, among others.
+Examples of shared functionality are _push_ on containers and on packages _build_.
```

Comment:
"Examples of shared functionality are _push_ on containers or _build_ on packages."